spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/tunnel/client/TunnelClient.java

```diff
@@ -85,13 +85,14 @@ public void afterSingletonsInstantiated() {
 	public int start() throws IOException {
 		synchronized (this.monitor) {
 			Assert.state(this.serverThread == null, "Server already started");
-			ServerSocketChannel serverSocketChannel = ServerSocketChannel.open();
-			serverSocketChannel.socket().bind(new InetSocketAddress(this.listenPort));
-			int port = serverSocketChannel.socket().getLocalPort();
-			logger.trace("Listening for TCP traffic to tunnel on port " + port);
-			this.serverThread = new ServerThread(serverSocketChannel);
-			this.serverThread.start();
-			return port;
+			try (ServerSocketChannel serverSocketChannel = ServerSocketChannel.open()) {
```

Comment:
This will close the `ServerSocketChannel` before it can be used by the `ServerThread`. The channel is eventually closed when the thread is closed so this change is unnecessary. I'll address this when merging.